File: works/2026-09-20-no-such-day/census/census.rb
# Four readers from one class. Ruby's Date takes a calendar-reform date as an
# argument; ITALY (1582-10-15) is the one you get when you do not pass one.
require 'date'
Y0, Y1 = 1500, 1930
out = ARGV[0]
{ 'ruby-default-italy' => Date::ITALY, 'ruby-england' => Date::ENGLAND,
  'ruby-julian' => Date::JULIAN, 'ruby-gregorian' => Date::GREGORIAN }.each do |name, start|
  File.open(File.join(out, name + '.txt'), 'w') do |f|
    (Y0..Y1).each do |y|
      (1..12).each do |m|
        (1..31).each do |d|
          begin
            f.puts Date.new(y, m, d, start).jd
          rescue Date::Error, ArgumentError
            f.puts '-'
          end
        end
      end
    end
  end
end
STDERR.puts "ruby #{RUBY_VERSION}"

File: works/2026-09-20-no-such-day/census/census.php
<?php
// Three readers in one binary: checkdate (strict, Gregorian), mktime (lenient),
// and juliantojd from the calendar extension - the Julian calendar, still shipped.
$Y0 = 1500; $Y1 = 1930; $out = $argv[1];
$a = fopen("$out/php-checkdate.txt", 'w');
$b = fopen("$out/php-mktime.txt", 'w');
$c = fopen("$out/php-juliantojd.txt", 'w');
for ($y = $Y0; $y <= $Y1; $y++)
  for ($m = 1; $m <= 12; $m++)
    for ($d = 1; $d <= 31; $d++) {
      fwrite($a, checkdate($m, $d, $y) ? gregoriantojd($m, $d, $y) . "\n" : "-\n");
      $ts = mktime(12, 0, 0, $m, $d, $y);
      fwrite($b, $ts === false ? "-\n" : (intdiv($ts - ($ts % 86400 + 86400) % 86400, 86400) + 2440588) . "\n");
      $j = juliantojd($m, $d, $y);
      fwrite($c, $j === 0 ? "-\n" : "$j\n");
    }
fwrite(STDERR, 'php ' . PHP_VERSION . "\n");
